sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -442,6 +459,13 @@ object SimplifyConditionals extends Rule[LogicalPlan] with PredicateHelper {
       case If(cond, trueValue, falseValue)
         if cond.deterministic && trueValue.semanticEquals(falseValue) => trueValue
 
+      case i @ If(cond, trueValue, falseValue) => cond match {
+        // If the null-check is redundant, remove it
+        case IsNull(child) if isRedundantNullCheck(trueValue, falseValue, child) => falseValue
+        case IsNotNull(child) if isRedundantNullCheck(falseValue, trueValue, child) => trueValue
+        case _ => i
+      }
```

Comment:
How about this format?;
```
      // If the null-check is redundant, remove it
      case If(IsNull(child), trueValue, falseValue)
        if isRedundantNullCheck(trueValue, falseValue, child) => falseValue
      case If(IsNotNull(child), trueValue, falseValue)
        if isRedundantNullCheck(falseValue, trueValue, child) => trueValue
```

Reply:
Ok

Reply:
Why did you add the inner pattern-matching (`cond match {` )? I think its better to avoid unnecessary pattern matching (In the current fix, all the cases for `If` exprs can be matched in the line 466).

Reply:
I see. You are right, i do not need the inner pattern match, i will fix that.